File: memcache_client_spring/src/main/java/com/example/memcache_client_spring/controller/AppController.java
package com.example.memcache_client_spring.controller;

import com.example.memcache_client_spring.service.MemcachedService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class AppController {
    @Autowired
    private MemcachedService memcachedService;

    @PostMapping("/set/{key}")
    public String setKeyValue(@PathVariable String key, @RequestParam String value) {
        memcachedService.setValue(key, value, 3600);
        return String.format("Key '%s' set successfully in Memcached", key);
    }

    @GetMapping("/get/{key}")
    public Object getKeyValue(@PathVariable String key) {
        return memcachedService.getValue(key);
    }
}

File: memcache_client_nest/src/app.controller.ts

import { Controller, Get, Post, Param, Query } from '@nestjs/common';
import { MemcachedService } from './memcached/memcached.service';

@Controller()
export class AppController {
  constructor(private readonly memcachedService: MemcachedService) {}

  @Post('/set/:key')
  async setKeyValue(
    @Param('key') key: string,
    @Query('value') value: string,
  ): Promise<any> {
    await this.memcachedService.setValue(key, value, 3600); // Set value with expiration of 1 hour (3600 seconds)
    return { message: `Key '${key}' set successfully in Memcached` };
  }

  @Get('/get/:key')
  async getKeyValue(@Param('key') key: string): Promise<any> {
    const value = await this.memcachedService.getValue(key);
    return { key, value };
  }
}
